### PR TITLE
Us003 view capability on role

### PR DESCRIPTION
**Pull request by CallumE74**
Screenshot 2023-09-22 at 14 27 40
Screenshot 2023-09-22 at 16 08 24

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,4 +36,5 @@ In the package.json, you can see that ESLint is running a check on the code afte
 ---
 
 When running pull request to main branch, the deploy.yml file will be run which will deploy the UI to the AWS Server.
-The link to the server is https://wvrhehxy3p.eu-west-1.awsapprunner.com/
\ No newline at end of file
+The link to the server is https://wvrhehxy3p.eu-west-1.awsapprunner.com/
+>>>>>>> main
```
Comment (sammyperry101) [inline README.md:40]: this main line is from merge conflict, shouldnt be here

File: src/model/JobRole.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export class JobRoleViewRoles{
```
Comment (TomBillingtonUK) [inline src/model/JobRole.ts:1]: should be export type

File: tsconfig.json [hunk 3]
```diff
@@ -6,7 +6,9 @@
     "forceConsistentCasingInFileNames": true,  
     "strict": true,                                    
     "noImplicitAny": true,
-    "esModuleInterop": true
+    "esModuleInterop": true,
+    "strictPropertyInitialization": false,
```
Comment (TomBillingtonUK) [inline tsconfig.json:10]: turn rule back on

File: tsconfig.json [hunk 4]
```diff
@@ -6,7 +6,9 @@
...
+    "allowSyntheticDefaultImports": true
```
Comment (TomBillingtonUK) [inline tsconfig.json:11]: remove this one

File: src/views/ViewRoles.html [hunk 5]
```diff
@@ -1,22 +1,51 @@
...
+                        <th scope="col" class="text-center">Sharepoint</th>
                     </tr>
-                {% endfor %}
-            </tbody>
-        </table>
+                </thead>
+                <tbody>
+                    {% for role in roles %}
+                        <tr>
+                            <th scope="row" class="text-center">{{role.roleName}}</a></td>
```
Comment (TomBillingtonUK) [inline src/views/ViewRoles.html:27]: the way the role names read in the list makes it hard to read so i would left align the text

File: test/unit/service/JobRoleServiceTests.ts [hunk 6]
```diff
@@ -31,15 +31,15 @@ describe('JobRoleService', function () {
...
             mock.onGet(process.env.BACK_URL + '/api/job-roles').reply(500);
 
-            let error = ''
+            let errorMessage = "default"
 
             try{
                 await viewJobRoles()
-            } catch (e) {
-                error = (e as Error).message;
+            } catch (e: any) {
```
Comment (TomBillingtonUK) [inline test/unit/service/JobRoleServiceTests.ts:38]: Fix this lint warning please

File: src/views/ViewRoles.html [hunk 7]
```diff
@@ -1,22 +1,51 @@
...
+                            <td class="text-center">{{role.bandName}}</td>
+                            <td class="text-center">{{role.capabilityName}}</td>
+                            <td class="text-center"> 
+                            <div class="btn-group mr-3" role="group">
+                                <button type="button" class="btn btn-primary" alt="View {{role.roleName}}">View</button>
+                            </div>
+                            </td>
+                            <td class="text-center">
+                                <!--<div class="btn-group mr-3" role="group">
```
Comment (TomBillingtonUK) [inline src/views/ViewRoles.html:36]: commented out code

**Review by TomBillingtonUK — APPROVED** [review]

**Review by sammyperry101 — APPROVED** [review]

**Pull request merged**